test/unit/krane/deploy_task_test.rb

```diff
@@ -23,4 +23,17 @@ def test_initializer_without_valid_file
     assert_logs_match("Configuration invalid")
     assert_logs_match(/File (\S+) does not exist/)
   end
+
+  def test_kubeconfig_configured_correctly
+    task = Krane::DeployTask.new(
+      namespace: "something",
+      context: KubeclientHelper::TEST_CONTEXT,
+      logger: logger,
+      current_sha: "",
+      filenames: ["unknown"],
+      kubeconfig: '/some/path.yml',
+    )
+    assert_equal('/some/path.yml', task.task_config.kubeconfig)
+    assert_equal(['/some/path.yml'], task.send(:kubeclient_builder).kubeconfig_files)
```

Comment:
Really don't love this here since we're testing a private method (and in the other couple of test files that do this). This smells that we're not exposing an API that we should be. I think making task_config the source of truth for kubeclient_builder obejcts as mentioned above will solve this.

Reply:
After moving the method to kubeconfig I added the test to the kubeconfig test, sans the send call. I wasn't sure whether to fully remove the other tests as it feels like there's a small bit of value in instantiating and checking the config, but I might have been overthinking things.